### PR TITLE
feat: checkout correct ref on pr

### LINKED ISSUE #72: CI checks out master branch on PR instead of PR branch
With the move to `pull_request_target` as the github workflow trigger, it's now checking out master in the CI pipeline instead of the PR branch

### PR DESCRIPTION
**Pull request by JQrdan**
Closes: #72

Signed-off-by: Jordan <jordan.tucker1@ibm.com>

### CODE REVIEW

**Comment by nictownsend** [comment]
LGTM and https://github.com/JQrdan/strimzi-ui/pull/5 shows the action that runs on base repo as working

**Review by nictownsend — APPROVED** [review]

File: .github/workflows/node-pr-jobs-secure.yml
Comment (matthew-chirgwin) [inline .github/workflows/node-pr-jobs-secure.yml:16]: Is it a guarantee we will always have jq from `ubuntu-latest`?

File: .github/workflows/node-pr-jobs.yml
Comment (matthew-chirgwin) [inline .github/workflows/node-pr-jobs.yml:5]: I assume this is seperate so `pull_request` refers to ref/branch being PR'ed - ie no need to HEAD as per above? I assume there is no way that information could be discovered once and shared between jobs (so we only have one workflow)?

**Review by matthew-chirgwin — COMMENTED** [review]
Looks good - and good spot in the first instance! Have some questions about the change, but none I would reject the PR on. Will also ask for a review from the maintainers/Pete/Don

**Review by pmuir — APPROVED** [review]
I've not used github actions much, generally 👍

File: .github/workflows/node-pr-jobs-secure.yml
Comment (JQrdan) [reply .github/workflows/node-pr-jobs-secure.yml:16]: Good point, might be worth changing these to an LTS release

File: .github/workflows/node-pr-jobs.yml
Comment (JQrdan) [reply .github/workflows/node-pr-jobs.yml:5]: Yep! I don't see the harm in these ones being normal `pull_request` workflow triggers as they don't need any secrets. It _might_ be possible to share the pr_head info as an artefact but given my previous fun with artefact downloading, I thought it would be best to leave that for now.

**Comment by JQrdan** [comment]
> I've not used github actions much, generally 👍

We've only played around with them slightly in the past but really liked them. Hence the teething problems with different GitHub topologies :D

**Review by matthew-chirgwin — CHANGES_REQUESTED** [review]
> Good point, might be worth changing these to an LTS release

Any plans to make a change to do this?

> Yep! I don't see the harm in these ones being normal `pull_request` workflow triggers as they don't need any secrets. It _might_ be possible to share the pr_head info as an artefact but given my previous fun with artefact downloading, I thought it would be best to leave that for now.

In a perfect world, we would have just the one file. I do wonder if two files will scale/last, but will not ask for early optimisation without a good use case. As a blocker however - it looks like e2es, storybook and linting are run in both `pull_request_target` and `pull_request` events. If the jobs are running repeatedly, can they be only run on one? There is a limit on the number of actions that can run at one time across the whole org I believe - we do not want to be a bad actor and waste resources.

**Review by scholzj — APPROVED** [review]

**Comment by JQrdan** [comment]
> In a perfect world, we would have just the one file. I do wonder if two files will scale/last, but will not ask for early optimisation without a good use case. As a blocker however - it looks like e2es, storybook and linting are run in both `pull_request_target` and `pull_request` events. If the jobs are running repeatedly, can they be only run on one? There is a limit on the number of actions that can run at one time across the whole org I believe - we do not want to be a bad actor and waste resources.

@matthew-chirgwin these are being duplicated in this PR as they have been defined as `pull_request` in this PR but are `pull_request_target` in master so there's a clash. Once merged, that will go away

**Review by matthew-chirgwin — APPROVED** [review]

**Pull request merged**
